src/weekly/query.js

```diff
@@ -17,42 +17,52 @@ async function queryZuora (deliveryDate, config: Config) {
     {
       'name': 'WeeklySubscriptions',
       'query': `
-        SELECT
-            RateplanCharge.quantity,
-            Subscription.Name,
-              SoldToContact.Address1,
-              SoldToContact.Address2,
-              SoldToContact.City, 
-              SoldToContact.Company_Name__c,
-              SoldToContact.Country, 
-              SoldToContact.Title__c,
-              SoldToContact.FirstName, 
-              SoldToContact.LastName, 
-              SoldToContact.PostalCode, 
-              SoldToContact.State,
-              Subscription.CanadaHandDelivery__c
-          FROM
-            rateplancharge
-          WHERE
-           (Subscription.Status = 'Active' OR Subscription.Status = 'Cancelled') AND
-           ProductRatePlanCharge.ProductType__c is not null AND
-           Product.Name like '%Weekly%' AND
-           RatePlanCharge.EffectiveStartDate <= '${formattedDate}' AND
-           RatePlanCharge.EffectiveEndDate >= '${formattedDate}' AND
-           RatePlan.AmendmentType != 'RemoveProduct'`
+      SELECT
+      RateplanCharge.quantity,
+      Subscription.Name,
+      SoldToContact.Address1,
+      SoldToContact.Address2,
+      SoldToContact.City,
+      SoldToContact.Company_Name__c,
+      SoldToContact.Country,
+      SoldToContact.Title__c,
+      SoldToContact.FirstName,
+      SoldToContact.LastName,
+      SoldToContact.PostalCode,
+      SoldToContact.State,
+      Subscription.CanadaHandDelivery__c,
+      Subscription.AutoRenew,
+      Subscription.Status,
+      ProductRatePlanCharge.ProductType__c,
+      Product.Name,
+      RatePlanCharge.EffectiveStartDate,
+      RatePlanCharge.EffectiveEndDate,
+      RatePlan.AmendmentType,
+      Subscription.TermEndDate
+        FROM
+          rateplancharge
+        WHERE
+         (Subscription.Status = 'Active' OR Subscription.Status = 'Cancelled') AND
+         ProductRatePlanCharge.ProductType__c = 'Guardian Weekly' AND
+         RatePlanCharge.EffectiveStartDate <= '${formattedDate}' AND
+         (
+          (Subscription.AutoRenew = true AND RatePlanCharge.EffectiveEndDate >= '${formattedDate}') OR
+          (Subscription.AutoRenew = false AND Subscription.TermEndDate >= '${formattedDate}')
+         ) AND
+         RatePlan.AmendmentType != 'RemoveProduct'`
```

Comment:
do we need this last part: 
 RatePlan.AmendmentType != 'RemoveProduct'`
?

Reply:
it's there for homedelivery, it's to avoid fulfilling to cancelled customers- but I'll double check it in the next query change

Reply:
are you sure it's there for HD ? 
https://github.com/guardian/fulfilment-lambdas/blob/master/src/homedelivery/query.js
i think it is there just for the suspensions query